src/Farmer/Builders/Builders.ContainerGroups.fs

```diff
@@ -406,7 +413,12 @@ type NetworkProfileBuilder() =
     member _.AddIpConfigs(state:NetworkProfileConfig, configs) = { state with ContainerNetworkInterfaceConfigurations = state.ContainerNetworkInterfaceConfigurations @ configs }
     /// Sets the virtual network for the profile
     [<CustomOperation "vnet">]
-    member _.VirtualNetwork(state:NetworkProfileConfig, vnet) = { state with VirtualNetwork = ResourceName vnet }
+    member _.VirtualNetwork(state:NetworkProfileConfig, vnet) =
+        { state with VirtualNetwork = Managed(virtualNetworks.resourceId(ResourceName vnet)) }
+    /// Links to an existing vnet.
+    [<CustomOperation "link_to_vnet">]
+    member _.LinkToVirtualNetwork(state:NetworkProfileConfig, vnet) =
```

Comment:
Just passing a name probably won't always be sufficient: if the vnet is in another resource group, ARM won't accept this.

In such a case, you should have an overload to supply a full `resourceId`, which can be created with the resource group of, in this case, the vnet.